packages/xstate-test/src/machine.ts

```diff
@@ -0,0 +1,135 @@
+import { serializeState, SimpleBehavior } from '@xstate/graph';
+import {
+  ActionObject,
+  AnyEventObject,
+  AnyState,
+  AnyStateMachine,
+  createMachine,
+  EventFrom,
+  EventObject,
+  StateFrom,
+  TypegenConstraint,
+  TypegenDisabled
+} from 'xstate';
+import { TestModel } from './TestModel';
+import {
+  TestMachineConfig,
+  TestMachineOptions,
+  TestModelEventConfig,
+  TestModelOptions
+} from './types';
+import { flatten } from './utils';
+import { validateMachine } from './validateMachine';
+
+export async function testStateFromMeta(state: AnyState) {
+  for (const id of Object.keys(state.meta)) {
+    const stateNodeMeta = state.meta[id];
+    if (typeof stateNodeMeta.test === 'function' && !stateNodeMeta.skip) {
+      await stateNodeMeta.test(state);
+    }
+  }
+}
+
+export function createTestMachine<
+  TContext,
+  TEvent extends EventObject = AnyEventObject,
+  TTypesMeta extends TypegenConstraint = TypegenDisabled
+>(
+  config: TestMachineConfig<TContext, TEvent, TTypesMeta>,
+  options?: TestMachineOptions<TContext, TEvent, TTypesMeta>
+) {
+  return createMachine(config, options as any);
+}
+
+export function executeAction(
+  actionObject: ActionObject<any, any>,
+  state: AnyState
+): void {
+  if (typeof actionObject.exec === 'function') {
+    actionObject.exec(state.context, state.event, {
+      _event: state._event,
+      action: actionObject,
+      state
+    });
+  }
+}
+
+/**
+ * Creates a test model that represents an abstract model of a
+ * system under test (SUT).
+ *
+ * The test model is used to generate test plans, which are used to
+ * verify that states in the `machine` are reachable in the SUT.
+ *
+ * @example
+ *
+ * ```js
+ * const toggleModel = createModel(toggleMachine).withEvents({
+ *   TOGGLE: {
+ *     exec: async page => {
+ *       await page.click('input');
+ *     }
+ *   }
+ * });
+ * ```
+ *
+ * @param machine The state machine used to represent the abstract model.
+ * @param options Options for the created test model:
+ * - `events`: an object mapping string event types (e.g., `SUBMIT`)
+ * to an event test config (e.g., `{exec: () => {...}, cases: [...]}`)
+ */
+export function createTestModel<TMachine extends AnyStateMachine>(
```

Comment:
`TestModel` is abstract and capable of working with simple behaviors, this function here (that is named `createTestModel`) is strongly coupled to machines though. We should consider renaming this to smth like `createMachineTestModel` or smth.

Reply:
I'm comfortable leaving this the way it is. I don't think we're going to focus on other behaviours in this major version (I don't see the point in it atm) so we should assume that all users of `@xstate/test` are going to be using machines as test models.

If they want to use something else as a test model, they can use `new TestModel` directly.